docs/billing.md

```diff
@@ -1,11 +1,84 @@
+1. Router: Non changing initial contract that users/dApps call to send a message
+2. OnRamp: Forwards Router's send request
+3. FeeQuoter: OnRamp calls to estimate the fees for sending a message <<< Important
+4. OffRamp: Receives the messages (Offchain part plays the part here to send the message to the other chain's OffRamp)
+
+So if sourceChain is sending a message to destinationChain the flow looks like
+
+1. sourceChain: Router -> OnRamp -> FeeQuoter
+2. OffChain using OffChain Reporting 3 (OCR3) processes send events emitted from OnRamp in Commit Plugin send a transaction commit on destinationChain's OffRamp
+3. destinationChain: OffRamp ---ocr3 commit-plugin--> FeeQuoter
+
+## OffChain
+
+We have 2 plugins (Commit and Execute).
+
+Commit is the one responsible for reporting gas prices, and token prices that don't come from keystone.
+
+Execute is responsible for fee boosting [Add Link] during the actual execution of messages.
+
+## Fee Structure
+
+To send a message from sourceChain to destinationChain we need to account for multiple fees. For in detailed doc check [billing](https://docs.chain.link/ccip/billing)
+
+1. Network/Premium fees.
+2. destinationChain Transaction fees (execution costs + data availability cost on the destination chain)
+
+Of-course the details of the calculation will depend on the message being sent and whether it has tokens to send or not, data availability..etc.
+For in details look on how the fees is calculated you can check FeeQuoter's [`getValidatedFee`](https://github.com/smartcontractkit/chainlink/blob/37f3132362ec90b0b1c12fb1b69b9c16c46b399d/contracts/src/v0.8/ccip/FeeQuoter.sol#L498)
+
+To be able to pay in one of the available fee tokens on sourceChain we need to arrive at a quote in USD and convert the USD amount into corresponding fee token amount.
+
+So the components we need to calculate the final price are:
+1. sourceChain fee token price, usually LINK and the native token of the chain. This is what the user pays in the end. TokenPriceProcessor will update them 
+2. destinationChain Fee/Gas price. FeeChainProcessor will update them.
+3. destinationChain native token price (to be able to calculate the fees denominated in USD
+
+
+## Token Prices Processor
+
+During commit round. Token prices are updated by TokenPriceProcessor. 
+
+For each round these are the steps:
+1. **Observation:**  
+   a. Fetches the token prices from USD feed for tokens we [configure](https://github.com/smartcontractkit/chainlink-ccip/blob/f03ff5183eb8323ba8e0a13dc58d1da13b755307/pluginconfig/commit.go#L89-L91) during the plugin initiation.   
+   b. Fetches current token prices stored in **destination chain FeeQuoter** (the chain the current node is supposed to commit to).
+2. **Outcome:**  
+Cross-check values from 1a and 1b. and posts the tokens that needs updating in the Outcome. The prices from the feed (1a) will be used when:  
```

Comment:
Does this mean the tokenPrice from the FeeQuoter is used in other situations? Or what happens if FeeQuoter and Feed both have price?

Is there a situation when the Feed is unavailable?

Reply:
If FeeQuoter has price that don't lie in one of the 3 categories mentioned will not be updated.


> Is there a situation when the Feed is unavailable?

We don't account for this. This is an offchain config that's injected during plugin initialization and need to be present before the plugin starts.